csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -754,247 +786,157 @@ std::vector<std::vector<MatmulDimRole>> HopperMultipleMatmulScheduler::
-    //
-    // Without split-K:
-    //   mma_result
-    //     nbatch +   1   2    3    4    5   6   7    8
-    //              -14 -13  -12  -11  -10  -9  -8   -7
-    //     [... iMo iNo rKg rKwo iMwo iNwo iMw iNw iMino
-    //    (iBz) iBx iBy  rS   rS  iTz  iTy  iS  iS  iTx
-    //                                   9    10    11     12    13    14
-    //                                  -6    -5    -4     -3    -2    -1
-    //                               iNino iMin2 iNin2  rKino rKin4 rKin2]
-    //                                iMMA  iMMA  iMMA   rMMA  rMMA  rMMA
-    //   smem_epilogue   (unscheduled, same as original mma_result)
-    //   splitk_sum
-    //     [... iMo iNo rKf  iMi  iNi]
-
-    // parallelize Mwo, Nwo by thread
-    mma_result->axis((int64_t)merged_roles.size() + num_splitk_dims_ + 1)
-        ->parallelize(ParallelType::TIDz);
-    mma_result->axis((int64_t)merged_roles.size() + num_splitk_dims_ + 2)
-        ->parallelize(ParallelType::TIDy);
-
-    if (params_->use_smem_epilogue) {
-      smem_epilogue->setMemoryType(MemoryType::Shared);
-      auto swizzled_dom = swizzleSharedMemory<false>(smem_epilogue);
-      smem_epilogue->setAllocationDomain(swizzled_dom.as<IterDomain*>(), true);
-      scheduler_utils::BoundedDirectionalTransformPropagator::forward(
-          mma_result,
-          -1,
-          {smem_epilogue},
-          scheduler_utils::BoundedDirectionalTransformPropagator::Options()
-              .propagateParallelType()
-              .propagateToBoundary());
-      smem_epilogue->axis(-1)->parallelize(ParallelType::Vectorize);
-    }
-
-    // When we have both batch dims and splitk, parallelize splitk only.
-    // If we only have batch dim, parallelize the batch dim.
-    if (params_->splitk_factor > 1) {
-      mma_result->axis(num_device_and_batch_dims_ + 2)
-          ->parallelize(ParallelType::BIDz);
-    } else if (num_local_batch_dims_ > 0) {
-      mma_result->axis(num_device_dims_)->parallelize(ParallelType::BIDz);
-    }
-    switch (params_->cta_order) {
-      case MatmulParams::TileRasterizationOrder::RowMajor:
-        mma_result->axis(num_device_and_batch_dims_)
-            ->parallelize(ParallelType::BIDx);
-        mma_result->axis(num_device_and_batch_dims_ + 1)
-            ->parallelize(ParallelType::BIDy);
-        break;
-      case MatmulParams::TileRasterizationOrder::ColumnMajor:
-        mma_result->axis(num_device_and_batch_dims_)
-            ->parallelize(ParallelType::BIDy);
-        mma_result->axis(num_device_and_batch_dims_ + 1)
-            ->parallelize(ParallelType::BIDx);
-        break;
-      default:
-        NVF_THROW("Invalid TileRasterizationOrder passed to Matmul scheduler");
-    }
+    mma_result->axis(-1)->parallelize(ParallelType::Mma);
+    mma_result->axis(-2)->parallelize(ParallelType::Mma);
+    mma_result->axis(-3)->parallelize(ParallelType::Mma);
   }
 }
 
 void HopperMultipleMatmulScheduler::schedulePrologues() {
```

Comment:
Do we need the definition of`schedulePrologues`? It doesn't exist in the header.

Reply:
We dont need the method. I just forgot to remove it.